cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/templates/etl/transforms/ProjectionTransform.java

```diff
@@ -0,0 +1,301 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.templates.etl.transforms;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.dataset.lib.KeyValue;
+import co.cask.cdap.templates.etl.api.Emitter;
+import co.cask.cdap.templates.etl.api.Property;
+import co.cask.cdap.templates.etl.api.StageConfigurer;
+import co.cask.cdap.templates.etl.api.Transform;
+import co.cask.cdap.templates.etl.api.TransformContext;
+import co.cask.cdap.templates.etl.common.KeyValueListParser;
+import com.google.common.base.Splitter;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+/**
+ * Projection transform that allows dropping, renaming, and converting field types.
+ */
+public class ProjectionTransform extends Transform<StructuredRecord, StructuredRecord> {
+  private static final String DROP = "drop";
+  private static final String RENAME = "rename";
+  private static final String CONVERT = "convert";
+  private static final Pattern fieldDelimiter = Pattern.compile("\\s*,\\s*");
+  private Set<String> fieldsToDrop = Sets.newHashSet();
+  private Map<String, String> fieldsToRename = Maps.newHashMap();
+  private Map<String, Schema.Type> fieldsToConvert = Maps.newHashMap();
```

Comment:
this one doesn't need to be a bimap since multiple fields can be converted to a string for example. but will check that each key only shows up once when initializing